[SecuritySolution][Navigation]  V2 sidenav fixes and udpates

ashokaditya opened the pull request:
## Summary
Fixes the issue where Translated Dashboards was not visible.

Follow up of https://github.com/elastic/kibana/pull/235048
See https://github.com/elastic/kibana/pull/235048#issuecomment-3317966366

<details open><summary>v2</summary>
<img width="1921" height="1218" alt="Screenshot 2025-09-24 at 11 38 24" src="https://github.com/user-attachments/assets/159561c8-8dfa-48a3-8467-82d7d0071e82" />
<img width="569" height="1032" alt="Screenshot 2025-09-24 at 13 20 03" src="https://github.com/user-attachments/assets/1d8b0223-7e00-4916-979b-6acf4e296f4f" />

</details> 

<details><summary>v1 and v2</summary>
<img width="1279" height="1166" alt="Screenshot 2025-09-24 at 12 21 57" src="https://github.com/user-attachments/assets/fe1d3836-02b8-4cef-b113-8c8088e402c2" />
</details> 

<details><summary>clip (v1 and v2 launchpad)</summary>
<p>
  <img src="https://github.com/user-attachments/assets/b6e3206f-d4ea-470c-a042-0a738ed4212b" />
</p>
</details> 

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

ashokaditya committed:
show translated dashboards on ECH v2 nav

refs https://github.com/elastic/kibana/pull/235048

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

ashokaditya committed:
show translated dashboards on serverless v2 nav

refs 61fc34e0738fa8acd7603d211e6048ccf6e28bc8

ashokaditya committed:
ensure links are correctly active for launchpad

As `Security launchpad` in v1 uses "items" for it's two links (get started, value report), instead of "groups" of items in v2 Launchpad where some items are siblings of nested children (get started, value report are siblings of migrations group that have translated rules/dashboards), when running both navs in tandem with `projectSideNav: both` the v1 links are not set active. This commit fixes that by ensuring that the v1 links are also groups of items

ashokaditya committed:
add workflows to nav v2

was added in https://github.com/elastic/kibana/pull/232629

semd reviewed (approved):
LGTM!

logeekal reviewed (approved):
I did desk testing in following scenarios from the perspective of Automatic Migrations and sidenav looks correct 🚀 . Thank you.

### ECH & Security Solution View

- [x] Translated Rules in Launchpad
- [x] Translated Dashboards in Launchpad


### ECH & Classic View

- [x] Translated Rules in Migrations
- [x] Translated Dashboards in Migrations


### Serverless & Security Project 

- [x] Translated Rules in Launchpad
- [x] Translated Dashboards in Launchpad

ashokaditya committed:
fix test

refs eb1d09653319ba7aafc37c84d9559955e89e08f5

ashokaditya committed:
update workflow icon

e40pud reviewed (approved):
LGTM! Also tested locally and I can now see `Translated dashboards` in the `Launchpad`. Thanks for the fix 🚀

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343240)
* Commit: 8005b92f9406c9abf1f715f6f1414edc55220d28

### Failed CI Steps
* [FTR Configs #60](https://buildkite.com/elastic/kibana-pull-request/builds/343240#01997c34-42af-4f82-b5e5-116fba2dc1bc)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/4cfcababfa2fc6a7606bc2bb8f6fd2718e128c2c) | [after](https://github.com/elastic/kibana/commit/8005b92f9406c9abf1f715f6f1414edc55220d28) | diff |
| --- | --- | --- | --- |
| `securitySolutionEss` | 115 | 116 | +1 |
| `securitySolutionServerless` | 150 | 151 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4cfcababfa2fc6a7606bc2bb8f6fd2718e128c2c) | [after](https://github.com/elastic/kibana/commit/8005b92f9406c9abf1f715f6f1414edc55220d28) | diff |
| --- | --- | --- | --- |
| `securitySolutionEss` | 37.5KB | 38.4KB | +978.0B |
| `securitySolutionServerless` | 61.1KB | 62.1KB | +996.0B |
| total |  |  | +1.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/4cfcababfa2fc6a7606bc2bb8f6fd2718e128c2c) | [after](https://github.com/elastic/kibana/commit/8005b92f9406c9abf1f715f6f1414edc55220d28) | diff |
| --- | --- | --- | --- |
| `securitySolutionEss` | 39.0KB | 39.3KB | +332.0B |
| `securitySolutionServerless` | 55.9KB | 56.2KB | +315.0B |
| total |  |  | +647.0B |

<details><summary>Unknown metric groups</summary>

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/4cfcababfa2fc6a7606bc2bb8f6fd2718e128c2c) | [after](https://github.com/elastic/kibana/commit/8005b92f9406c9abf1f715f6f1414edc55220d28) | diff |
| --- | --- | --- | --- |
| `securitySolutionEss` | 8 | 9 | +1 |
| `securitySolutionServerless` | 21 | 22 | +1 |
| total |  |  | +2 |


</details>


### History
* :broken_heart: [Build #343094](https://buildkite.com/elastic/kibana-pull-request/builds/343094) failed 44ec56b7bd5e98e42c89b62d4929c3b76893ad05

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343240","number":343240,"commit":"8005b92f9406c9abf1f715f6f1414edc55220d28"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343094","number":343094,"commit":"44ec56b7bd5e98e42c89b62d4929c3b76893ad05"}],"number":343240}
buildkite-pr-comment-->
